### PR TITLE
Soback 19 & soback 20 export of search

### PR DESCRIPTION
**Pull request by jesperlauridsen**
Added option for search result export
 Added warc export option
 Added warcExpanded export option
Added csv export option
Added modal for selecting fields to be exported to csv
Styled the whole thing to look somewhat decent

Minor fix in searchbox function to check if anything was changed  to fire a new search. Wrong params were used in the check, weirdly enough.

Removed some console.logs, and did some general cleanup

### CODE REVIEW

**Review by jorntx — COMMENTED** [review]
Very good work!  I like the simplecity in the export solution. Very few comments and 2 issues:

1. You can't move the top item downwards in the export CSV function. All other toggling and adding works fine.

2. I have a weird graphical difference between the up and down arrow in the export CSV function

File: src/js/src/components/SearchResults/AllSearchResults.vue [hunk 1]
```diff
@@ -4,6 +4,7 @@
       <search-facet-options />
     </div>
     <div class="resultContainer">
+      <search-result-export v-if="configs" />
```
[the remark this answers is not in the document]
Comment (jesperlauridsen) [reply src/js/src/components/SearchResults/AllSearchResults.vue:7]: :+1:

File: src/js/src/components/SearchResults/SearchResultExport.vue [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+      <button v-if="configs.exportOptions.csvAllowed === 'true'" class="exportButton" @click="toggleCsvExportOptions()">
+        CSV export
+      </button>
+    </div>
+    <div v-if="csvExportOpen" class="csvExportOptions">
+      <div class="csvExportContent">
+        <button class="closeButton" @click="toggleCsvExportOptions()">
+          ✕ 
+        </button>
+        <div class="exportContent" />
```
[the remark this answers is not in the document]
Comment (jesperlauridsen) [reply src/js/src/components/SearchResults/SearchResultExport.vue:22]: No - its a placeholder div to to make sure i have the margin i want with the columns i want. It's a 5 column grid layout, with the first and fifth being empty, as it is now.

File: src/js/src/components/SearchResults/SearchResultExport.vue [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+      return fields.replace(/ /g, '').split(',').slice(0,9)
+    },
+    getSplitFieldsNotSelected(fields) {
+      let newArray = fields.replace(/ /g, '').split(',')
+      return newArray.slice(9,newArray.length)
+    },
+    moveItemInArray(array, direction, itemNumber, item) {
+      if(itemNumber >= 1 && itemNumber < array.length) {
+      array.splice(itemNumber, 1)
+      direction === 'up' ? array.splice(itemNumber - 1, 0, item) : array.splice(itemNumber + 1, 0, item)
```
[the remark this answers is not in the document]
Comment (jesperlauridsen) [reply src/js/src/components/SearchResults/SearchResultExport.vue:121]: :+1:

File: src/js/src/components/SearchResults/SearchResultExport.vue [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+    exportToExtendedWARC() {
+      return this.searchAppliedFacets ? 
+      configs.playbackConfig.solrwaybackBaseURL + 'services/export/warcExpanded?query=' + this.query + this.searchAppliedFacets :
+      configs.playbackConfig.solrwaybackBaseURL + 'services/export/warcExpanded?query=' + this.query
+    },
+    exportToCSV() {
+      let fields = this.selectedArray.join(',')
+      return this.searchAppliedFacets ? 
+      configs.playbackConfig.solrwaybackBaseURL + 'services/export/csv?query=' + this.query + this.searchAppliedFacets + '&fields=' + fields :
+      configs.playbackConfig.solrwaybackBaseURL + 'services/export/csv?query=' + this.query + '&fields=' + fields
```
[the remark this answers is not in the document]
Comment (jesperlauridsen) [reply src/js/src/components/SearchResults/SearchResultExport.vue:106]: :+1:

File: src/js/src/components/SearchResults/SearchResultExport.vue [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+      array.splice(itemNumber, 1)
+      direction === 'up' ? array.splice(itemNumber - 1, 0, item) : array.splice(itemNumber + 1, 0, item)
+      }
+    },
+    toggleItemInArrays(toArray, fromArray, item, itemNumber, recipient) {
+      recipient === 'nonSelectedArray' ? toArray.push(item) : toArray.unshift(item)
+      fromArray.splice(itemNumber,1)
+
+    }
+  },
```
[the remark this answers is not in the document]
Comment (jesperlauridsen) [reply src/js/src/components/SearchResults/SearchResultExport.vue:129]: :+1: With the bugfix it's a little more complex. not much tho!

**Comment by jesperlauridsen** [comment]
merged newfrontend into branch aswell as updated for fixes. Should be good to go now @jorntx !

**Comment by jorntx** [comment]
I still have a bug - _id_ will only move one place down then it jumps up again..

**Pull request merged**